src/super_gradients/training/transforms/detection/abstract_detection_transform.py

```diff
@@ -31,6 +31,14 @@ def apply_to_sample(self, sample: DetectionSample) -> DetectionSample:
         """
         raise NotImplementedError
 
+    @property
```

Comment:
I think this PR is almost ready to be accepted.
A one nuance I still exists, howerver.
In dataset class we have a sanity check:
```
        for transform in transforms:
            if load_sample_fn is None and transform.additional_samples_count > 0:
                raise RuntimeError(
                    f"Detected transform {transform.__class__.__name__} that require {transform.additional_samples_count} "
                    f"additional samples, but load_sample_fn is None"
                )
```

So this check may actuall not be performed if we sample 0.
Instead I suggest to change the base class and have a method and an attribute:
```
@property
def may_require_additional_samples -> bool:
   return False # default impl

def get_number_of_additional_samples() -> int:
   return 0 # Default impl
```

And make a note that `get_number_of_additional_samples` is not deterministic.

I think this would be a cleanest implementation.

The `additional_samples_count` property can use `get_number_of_additional_samples` implementation for backward-compatibility reasons but be deprecated

Reply:
Addressed, [4ba5698](https://github.com/Deci-AI/super-gradients/pull/1897/commits/4ba5698ba50f9946d956e1b6b76e560478f72264), [a6486c9](https://github.com/Deci-AI/super-gradients/pull/1897/commits/a6486c918874fc6b69d0d7110843628259bc1c29)